### DISCUSSION TITLE
SoapHeaderMiddleware Example not working because of type error - how to use correctly?

### DISCUSSION THREAD

**Discussion by trailsnail** [phpro/soap-client#542] [Q&A] [answered]
Good morning again 🌞 

I try to manipulate our soap header before sending the request. The [given example here](https://github.com/php-soap/psr18-transport/?tab=readme-ov-file#soapheadermiddleware) does not work because of type error:

```
Soap\Psr18Transport\Middleware\SoapHeaderMiddleware::__construct(): Argument #1 must be of type callable, SoapHeader given
```

Should I not use the Middleware or how it must be used?
Maybe I have to go the [other way here](https://github.com/php-soap/xml#soapheaders)?

**Comment by veewee** [comment, marked as answer]
It seems like you are injecting PHP's https://www.php.net/manual/en/class.soapheader.php instead of `Soap\Xml\Builder\SoapHeader`?

The provided example should work, given it is covered by unit tests:
https://github.com/php-soap/psr18-transport/blob/main/tests/Unit/Middleware/SoapHeaderMiddlewareTest.php
